src/lava/lib/optimization/solvers/generic/builder.py

```diff
@@ -0,0 +1,365 @@
+        return vars
+
+    @staticmethod
+    def _get_initial_value_for_var(coefficient: npt.ArrayLike,
+                                   rank: int) -> npt.ArrayLike:
+        """Get the value for initializing the coefficient's Var.
+
+        Parameters
+        ----------
+
+        coefficient: A tensor representing one of the coefficients of a cost
+        or constraints function.
+        rank: the rank of the tensor coefficient.
+
+        """
+        if rank == 1:
+            return - coefficient
+        if rank == 2:
+            quadratic_component = coefficient * np.logical_not(
+                np.eye(*coefficient.shape))
+            return - quadratic_component
+
+    @staticmethod
+    def _update_linear_component_var(vars: ty.Dict[int, AbstractProcessMember],
+                                     quadratic_coefficient: npt.ArrayLike):
+        """Update a linear coefficient's Var given a quadratic coefficient.
+
+        Parameters
+        ----------
+
+        vars: A dictionary where keys are ranks and values are the Lava Vars
+        corresponding to ranks' coefficients.
+        quadratic_coefficient: an array-like tensor of rank 2, corresponds to
+        the coefficient of the quadratic term on a cost or constraint function.
+
+        """
+        linear_component = -quadratic_coefficient.diagonal()
+        if 1 in vars.keys():
+            vars[1].init = vars[1].init + linear_component
+        else:
+            vars[1] = Var(shape=linear_component.shape, init=linear_component)
+
+    def _in_ports_from_coefficients(self,
+                                    coefficients: CoefficientTensorsMixin) -> \
+            ty.Dict[int, AbstractProcessMember]:
+        """Create input ports for the various ranks of a coefficients object.
+
+        Parameters
+        ----------
+
+        coefficients: A tensor representing one of the coefficients of a cost
+        or constraints function.
+
+        """
+        in_ports = {coeff.ndim: InPort(shape=coeff.shape) for coeff in
+                    coefficients.coefficients}
+        return in_ports
+
+    def _decorate_process_model(self,
+                                solver_model: "OptimizationSolverModel",
+                                requirements: ty.List[AbstractComputeResource],
+                                protocol: AbstractSyncProtocol) -> \
+            AbstractProcessModel:
+        """Replicate the functionality of the decorators for creating Lava
+        ProcessModels.
+
+        This is necessary when dynamically creating Lava process model classes.
+
+        Parameters
+        ----------
+        solver_model: A Lava process created as a solver from the specification
+        of an optimization problem.
+        requirements: specifies which resources the ProcessModel requires.
+        protocol: The SyncProtocol that the ProcessModel implements.
+
+        """
+        setattr(solver_model, "implements_process", self.solver_process)

```

Comment:
This is not really future-proof. At the moment, the decorators may not do more than setting attributes. But who knows if this will stay the same in the future? Thus, we should find a solution for this. Maybe add a git issue for now?